api/src/main/java/lab/en2b/quizapi/commons/user/UserService.java

```diff
@@ -26,8 +26,10 @@ public UserDetails loadUserByUsername(String email) throws UsernameNotFoundExcep
         return UserDetailsImpl.build(userRepository.findByEmail(email).orElseThrow(() -> new InvalidAuthenticationException("Invalid email or password provided!")));
     }
     public void createUser(RegisterDto registerRequest, String roleName){
-        if (userRepository.existsByEmail(registerRequest.getEmail()) || userRepository.existsByUsername(registerRequest.getUsername())) {
+        if (userRepository.existsByEmail(registerRequest.getEmail()) ) {
             throw new IllegalArgumentException("Error: email is already in use!");
+        }else if( userRepository.existsByUsername(registerRequest.getUsername())){
```

Comment:
This does not pass SonarCloud code analysis. The problem seems to be regarding null-checking the Boolean class objects returned by the repository.

Reply:
Solved ^^